src/ripple/app/tx/impl/NFTokenMint.cpp

```diff
@@ -146,6 +186,54 @@ NFTokenMint::preclaim(PreclaimContext const& ctx)
             return tecNO_PERMISSION;
     }
 
+    if (auto const amount = ctx.tx[~sfAmount])
+    {
+        if (hasExpired(ctx.view, ctx.tx[~sfExpiration]))
+            return tecEXPIRED;
+
+        if (!(ctx.tx.getFlags() & nft::flagCreateTrustLines) &&
+            !amount->native() && ctx.tx[~sfTransferFee])
+        {
+            auto const nftIssuer =
+                ctx.tx[~sfIssuer].value_or(ctx.tx[sfAccount]);
+
+            if (!ctx.view.exists(keylet::line(nftIssuer, amount->issue())))
```

Comment:
Due to a bug would cause failure if the NFT issuer is also the issuer of the IOU (if transferFee is present), this amendment would also need to this change

```
            if (issuer != amount.getIssuer() &&
                !ctx.view.read(keylet::line(issuer, amount.issue())))
                return tecNO_LINE;
```

This is fixed in another amendment [here](https://github.com/XRPLF/rippled/pull/4946/files). We would need to add test case for this as well.

Reply:
This Amendment has not yet been merged, so the code in `ctx.view.rules().enabled(fixNFTokenTrustlineSurprise)` cannot be used.

How can I do this?
Should this be the default process without `ctx.view.rules().enabled(fixNFTokenTrustlineSurprise)`?

Reply:
I don't think there is the need to check for the fixNFTokenTrustlineSurprise amendment at all.